File: csharp/models/block.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

public sealed class Block
{
    public Block(HashSet<Cell> cells, UInt16 row, UInt16 col)
    {
        Cells = cells;
        Row = row;
        Column = col;
        if (Row > 2)
        {
            throw new ArgumentOutOfRangeException("row", $"Row: {row} out of range");
        }
        if (Column > 2)
        {
            throw new ArgumentOutOfRangeException("col", $"Column: {col} out of range");
        }
    }
    public readonly HashSet<Cell> Cells;
    public readonly UInt16 Row;
    public readonly UInt16 Column;
}

File: libs/src/models/block.rs
 // use Set to assure unique cells per block

pub mod libsudoku {
    pub mod models {
        use std::collections::HashSet;

        use crate::models::cell::{libsudoku::models::Cell};

        pub struct Block {
            cells: HashSet<HashSet<Cell>>, // each row (or column) must be unique
            width: u8,                     // usually 3 x 3 cells to a block
            height: u8,
        }
        impl Block {
            fn new(cells: HashSet<HashSet<Cell>>, width: u8, height: u8) -> Block {
                let ret = Block {
                    cells,
                    width,
                    height,
                };
                ret.validate().unwrap(); // should panic if invalid
                ret
            }
            fn validate(&self) -> Result<(), &str> {
                if self.width > 3 {
                    return Err("Width cannot exceed 3");
                }
                if self.height > 3 {
                    return Err("Height cannot exceed 3");
                }
                Ok(())
            }
        }
    }
}
